chain/src/chain.rs

```diff
@@ -800,25 +836,33 @@ impl BlockChain {
     pub fn create_block_template_by_header(
         &self,
         author: AccountAddress,
-        previous_header: BlockHeader,
+        mut previous_header: BlockHeader,
         user_txns: Vec<SignedUserTransaction>,
         uncles: Vec<BlockHeader>,
         block_gas_limit: Option<u64>,
         tips: Option<Vec<HashValue>>,
     ) -> Result<(BlockTemplate, ExcludedTxns)> {
-        let current_number = previous_header.number().saturating_add(1);
         let epoch = self.epoch();
         let on_chain_block_gas_limit = epoch.block_gas_limit();
         let final_block_gas_limit = block_gas_limit
             .map(|block_gas_limit| min(block_gas_limit, on_chain_block_gas_limit))
             .unwrap_or(on_chain_block_gas_limit);
-        let (_, tips_hash) = if current_number <= self.dag_fork_height()?.unwrap_or(u64::MAX) {
-            (None, None)
-        } else if tips.is_some() {
-            (Some(self.get_block_dag_genesis(&previous_header)?), tips)
-        } else {
-            let result = self.current_tips_hash(&previous_header)?.expect("the block number is larger than the dag fork number but the state data doese not exis");
-            (Some(result.0), Some(result.1))
+        let tips_hash = match self.check_dag_type()? {
+            DagHeaderType::Single => None,
+            DagHeaderType::Genesis | DagHeaderType::Normal => {
```

Comment:
Genesis is a single chain block, but used as dag genesis. It's not a dag block. I think tips must be ignored for DagGenesis.

Reply:
I think check dag type can be removed. This can be modified in next version.

Reply:
So fix it right or remove it!